pages/courseDashboard.js

```diff
@@ -1,7 +1,37 @@
+import React, { useContext, useState, useEffect } from 'react'
+import { useAppContext } from '../context/app-context'
+import { getData } from '../context/apiCalls'
+import { calculations } from '../utilities/calculations'
+
 import styles from '../styles/Home.module.scss'
 
 export default function courseDashboard() {
+  const { sharedState, setSharedState } = useAppContext()
+  const courseId = sharedState.currentCourse
+  const [course, setCourse] = useState({})
+  const [activityTotals, setActivityTotals] = useState({})
+  const [percentageByMod, setPercentageByMod] = useState([])
+  const [percentageByActivity, setPercentageByActivity] = useState([])
```

Comment:
I think having these multiple local states is awesome! I assume they will be passed down to the chart components through props?

Reply:
Yes i think that was my intent. Hopefully that works out well, otherwise we can re-address